cpp/log/cert.cc

```diff
@@ -292,13 +292,14 @@ StatusOr<bool> Cert::HasBasicConstraintCATrue() const {
 
   const StatusOr<void*> ext_struct(ExtensionStructure(NID_basic_constraints));
 
-  if (!ext_struct.ok()) {
+  if (!ext_struct.ok()
+      && ext_struct.status().CanonicalCode() != Code::NOT_FOUND) {
     // Truly odd.
     LOG(ERROR) << "Failed to check BasicConstraints extension";
     return ext_struct.status();
   }
 
-  if (!ext_struct.ValueOrDie()) {
+  if (ext_struct.status().CanonicalCode() == Code::NOT_FOUND) {
```

Comment:
You could move this check above the one at L295, and then that one could revert to simply `if (!ext_struct.ok())`, might be a little easier to read.
Same for the other instance below at ~L322